server/routes/serviceProviderReferrals/interventionProgressView.ts

```diff
@@ -110,12 +110,24 @@ export default class InterventionProgressView {
           { text: `Session ${row.sessionNumber}` },
           { text: `${row.appointmentTime}` },
           { text: tagMacro(row.tagArgs as TagArgs) },
-          { html: `${row.linkHtml}` },
+          { html: this.linkHtml(row.links) },
         ]
       }),
     }
   }
 
+  private linkHtml(links: { text: string; href: string }[]): string {
+    const linkArray = links.map(link => {
+      if (link.text && link.href) {
+        return `<a class="govuk-link" href="${link.href}">${link.text}</a>`
```

Comment:
These both need a `ViewUtils.escape`.

Reply:
Even if there's no user input? These are coming from the urls / text fields we've hardcoded in the presenter.

Reply:
Yeah, what if there are non HTML-safe characters in the text, for example? They still need to be escaped.

Reply:
Ok, have updated this